internal/namespaces/vpcgw/v1/custom_gateway_network.go

```diff
@@ -35,3 +36,32 @@ func gatewayNetworkCreateBuilder(c *core.Command) *core.Command {
 	}
 	return c
 }
+
+var completeListPrivateNetworkCache *vpcgw.ListGatewayNetworksResponse
+
+func autocompletePrivateNetwork(ctx context.Context, prefix string) core.AutocompleteSuggestions {
+	suggestions := core.AutocompleteSuggestions(nil)
+
+	client := core.ExtractClient(ctx)
+	api := vpcgw.NewAPI(client)
+
+	if completeListPrivateNetworkCache == nil {
+		res, err := api.ListGatewayNetworks(&vpcgw.ListGatewayNetworksRequest{})
```

Comment:
Does the completion depends on the zone if it is filled up?

Reply:
Zone is unused for this type of completion, it will rely on default zone.
With fr-par-1 as default zone:
`scw vpc-gw gateway-network create zone=nl-ams-1 private-network-id=<tab>` will rely on fr-par-1

Reply:
Zone is now used